File: game_prototype/Player.py
import Move
import Statuses
import Stat
import Item
import GameEnum
import math
import copy
import Map

class Inventory():
    items = []
    def __init__(self):
        self.items = []
    def addItem(self, item):
        self.items.append(item)
    def removeItem(self, item):
        self.items.remove(item)

    def printItems(self):
        tempStr = ""
        for i in self.items:
            tempStr += str(i) + "\n"
        return tempStr


class Player:
    gold = 100
    location = [0, 0]
    stat = 0
    shielding = 0
    lastMoveInit = 0
    inventory = []
    moves = [Move.Move(GameEnum.MoveType.ATTACK, 2, 100,  name = "attack"), Move.Move(GameEnum.MoveType.DEFEND, 2, 75,  name = "defend"), Move.Move(GameEnum.MoveType.ATTACK, 4, 150, name = "strong attack")]
    initiative = 5
    def __init__(self, x, y):
        self.location = [x, y]
        self.stat = Stat.Stat()
        self.status = Statuses.Statuses()

    def addItem(self, item):
        self.inventory.append(item)
    
    def printInventory(self):
        tempStr = ""
        for i in self.inventory:
            tempStr += str(i) + "\n"
        return tempStr

    def getSpeed(self):
        return 1 - (self.stat.speed/100)
    
    def getAttack(self):
        tempAttack = self.stat.attack
        for i in self.inventory:
            tempAttack += i.stats.attack
        tempAttack = self.status.calculateAttack(tempAttack)
        return math.ceil(tempAttack)
    
    def getDefense(self):
        tempDef = self.stat.defense
        for i in self.inventory:
            tempDef += i.stats.defense
        
        if self.shielding:

            temp=  tempDef * self.shielding
            self.shielding = 0
            return temp
        
        return self.stat.defense
        
    def location(self):
        return self.location
    def moveLoc(self, dir, map):

        tempLoc = copy.copy(self.location)
        if dir == GameEnum.Direction.UP:
            tempLoc[1] -= 1
        elif dir == GameEnum.Direction.DOWN:
            tempLoc[1] += 1
        elif dir == GameEnum.Direction.LEFT:
            tempLoc[0] -= 1
        elif dir == GameEnum.Direction.RIGHT:
            tempLoc[0] += 1
        else:
            input("Invalid direction")
            return False
        return tempLoc
    
    def move(self, tempLoc, terrain):
        if GameEnum.terrain == GameEnum.Terrain.INVALID:
            print("Invalid move")
            return False
        else:
            print("Valid move")
            self.location = tempLoc
    def movesStr(self):
        tempStr = ""
        for i in self.moves:
            tempStr += i.name + " "
        return tempStr

    def getMove(self):
        while True:
            inputPlayer = input("Input Move Valid Moves:" + self.movesStr() + "\n")
            inputPlayer = inputPlayer.lower()

            for i in self.moves:
                if i.name == inputPlayer:
                    return i
            
            print("Invalid move")


if __name__ == "__main__":
    player = Player(0, 0)
    print(player.getAttack())
    player.moveLoc(GameEnum.Direction.UP, Map.Map())
    print(player.getDefense())

File: frontend/TodoFromScratch/app/src/main/java/com/example/todofromscratch/game/Player.java
package com.example.todofromscratch.game;


import com.example.todofromscratch.model.domain.Stat;
import com.example.todofromscratch.model.domain.Statuses;

import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;



// enum Stance {
//     ATTACK, // Add other stances as needed
// }

class Player {
    private Stances.StanceEnum stance = Stances.StanceEnum.ATTACK;
    private int gold;
    private int[] location;
    private Stat stat;
    private int shielding;
    private int lastMoveInit;
    private Inventory inventory;
    private List<Move> moves;
    private int initiative;
    private Statuses status;
    private Move nextMove;

    public Player(int x, int y) {
        this.location = new int[]{x, y};
        this.stat = new Stat();
        this.inventory = new Inventory();
        this.moves = new ArrayList<Move>();
        this.initiative = 5;
        this.status = new Statuses();

        // Initialize moves
        moves.add(new Move(GameEnum.MoveType.USEITEM, 0, 0, "Use an Item"));
        moves.add(new Move(GameEnum.MoveType.ATTACK, 2, 100, "attack"));
        moves.add(new Move(GameEnum.MoveType.DEFEND, 2, 75, "defend"));
        moves.add(new Move(GameEnum.MoveType.ATTACK, 4, 150, "strong attack"));
        moves.add(new Move(GameEnum.MoveType.SPELL, 4, 50,"5", "fireball"));
    }

    public void addItem(Item item) {
        inventory.add(item);
    }

    public Move getNextMove(){
        return nextMove;
    }

    public void setNextMove(Move nextMove){
        this.nextMove = nextMove;
    }


    public List<Move> getMoves(){
        return moves;
    }
    public void removeItem(Item item) {
        inventory.remove(item);
    }
    public int getGold() {
        return gold;
    }

    public void setGold(int gold) {
        this.gold = gold;
    }

    public void addGold(int gold) {
        this.gold += gold;
    }
    public Stat getStat() {
        return stat;
    }
    public void setStat(Stat stat) {
        this.stat = stat;
    }
    public void setHealth(int health) {
        stat.setHealth(health);
    }


    public String printInventory() {
        StringBuilder tempStr = new StringBuilder();
        for (Item item : inventory.getItems()) {
            tempStr.append(item).append("\n");
        }
        return tempStr.toString();
    }
    public Stances.StanceEnum getStance() {
        return stance;
    }
    public double getSpeed() {
        return 1 - (stat.getSpeed() / 100);
    }

    public int getAttack() {
        int tempAttack = stat.getAttack();
        for (Item item : inventory.getItems()) {
            tempAttack += item.getStats().getAttack();
        }
        tempAttack = status.calculateAttack(tempAttack); // Assuming 'status' is an instance variable of the Statuses class
        return (int) Math.ceil(tempAttack);
    }
    public Statuses getStatus() {
        return status;
    }
    public int getInitiative() {
        return initiative;
    }

    public void setMana(int mana) {
        stat.setMana(mana);
    }

    public int getMana() {
        return stat.getMana();
    }
    public Inventory getInventory() {
        return inventory;
    }

    public void setLastMoveInit(int lastMoveInit) {
        this.lastMoveInit = lastMoveInit;
    }

    public void setInitiative(int initiative) {
        this.initiative = initiative;
    }

    public int getLastMoveInit() {
        return lastMoveInit;
    }
    public int setShielding(int shield) {
        this.shielding = shield;
        return shield;
    }

    public int getDefense() {
        int tempDef = stat.getDefense();
        for (Item item : inventory.getItems()) {
            tempDef += item.getStats().getDefense();
        }

        if (shielding != 0) {
            int temp = tempDef * shielding;
            shielding = 0;
            return temp;
        }

        return stat.getDefense();
    }

    public int[] getLocation() {
        return location;
    }

    public int[] moveLoc(GameEnum.Direction dir, Map map) {
        int[] tempLoc = location.clone();
        switch (dir) {
            case UP:
                tempLoc[1] -= 1;
                break;
            case DOWN:
                tempLoc[1] += 1;
                break;
            case LEFT:
                tempLoc[0] -= 1;
                break;
            case RIGHT:
                tempLoc[0] += 1;
                break;
            default:
                System.out.println("Invalid direction");
                return null;
        }
        return tempLoc;
    }

    public void move(int[] tempLoc, GameEnum.Terrain terrain) {
        if (terrain == GameEnum.Terrain.INVALID) {
            System.out.println("Invalid move");
        } else {
            System.out.println("Valid move");
            location = tempLoc;
        }
    }

    public String movesStr() {
        StringBuilder tempStr = new StringBuilder();
        for (Move move : moves) {
            tempStr.append(move.getName()).append(", ");
        }
        return tempStr.substring(0, tempStr.length() - 2);
    }

    public Move getMove() {
        Scanner scanner = new Scanner(System.in);
        while (true) {
            System.out.println("Input Move Valid Moves:" + movesStr());
            String inputPlayer = scanner.nextLine().toLowerCase();

            for (Move move : moves) {
                if (move.getName().toLowerCase().equals(inputPlayer)) {
                    return move;
                }
            }

            System.out.println("Invalid move");
        }
    }

    public static void main(String[] args) {
        Player player = new Player(0, 0);
        System.out.println(player.getAttack());
        player.moveLoc(GameEnum.Direction.UP, new Map());
        System.out.println(player.getDefense());
        System.out.println(player.getLocation()[0]);
    
        System.out.println(player.movesStr());

    }
}
